libs/zjobs/src/zjobs.zig

```diff
@@ -478,23 +478,9 @@ pub fn JobQueue(
             if (prereqs.len == 0) return JobId.none;
             if (prereqs.len == 1) return prereqs[0];
 
-            var id = JobId.none;
-            var i: usize = 0;
-            const in: []const JobId = prereqs;
-            while (i < in.len) {
-                var job = CombinePrereqsJob{ .jobs = self };
-
-                // copy prereqs to job
-                var o: usize = 0;
-                const out: []JobId = &job.prereqs;
-                while (i < in.len and o < out.len) {
-                    out[o] = in[i];
-                    i += 1;
-                    o += 1;
-                }
-
-                id = try self.schedule(id, job);
-            }
+            var job = CombinePrereqsJob.init(self);
+            std.mem.copy(JobId, &job.prereqs, prereqs);
```

Comment:
What happens when `prereqs` is longer than `&job.prereqs`?  This is why I have a loop, it specifically creates a number of `CombinePrereqsJobs` to accomodate however many `prereqs` are supplied to this function.